zone/effects.cpp

```diff
@@ -509,6 +509,87 @@ bool Client::TrainDiscipline(uint32 itemid) {
 	return(false);
 }
 
+bool Client::MemorizeSpellFromItem(uint32 item_id) {
+	const EQ::ItemData *item = database.GetItem(item_id);
+	if(item == nullptr) {
+		Message(Chat::Red, "Unable to find the scroll!");
+		LogError("Unable to find scroll id [{}]\n", (unsigned long)item_id);
+		return false;
+	}
+
+	if (!item->IsClassCommon() || item->ItemType != EQ::item::ItemTypeSpell) {
+		Message(Chat::Red, "Invalid item type, you cannot learn from this item.");
+		SummonItem(item_id);
+		return false;
+	}
+
+	if(!(
```

Comment:
Not necessarily requesting changes here for this PR but converting the char array to a std::string would make it easy to do a string find and simplify the code a bit

Example

```cpp
		std::string item_name = item->Name;
		if (item_name.find("Spell: ") != std::string::npos) {
			std::cout << "found!" << '\n';
		}
```